### ISSUE TITLE
App crash on going back from a similar event

### ISSUE THREAD

**Issue by addiegupta** [fossasia/open-event-android#1474]
**Describe the bug**

The app crashes when a similar event is selected from `EventDetailsFragment` for an event and then the up button or the back button is pressed.

**To Reproduce**

The app should not crash

**Logs**

2019-03-30 15:50:59.520 6773-6824/com.eventyay.attendee E/AndroidRuntime: FATAL EXCEPTION: pool-1-thread-1
    Process: com.eventyay.attendee, PID: 6773
    java.lang.IndexOutOfBoundsException: Index: 27, Size: 0
        at java.util.ArrayList.get(ArrayList.java:437)
        at androidx.recyclerview.widget.AsyncListDiffer$1$1.areItemsTheSame(AsyncListDiffer.java:306)
        at androidx.recyclerview.widget.DiffUtil.diffPartial(DiffUtil.java:268)
        at androidx.recyclerview.widget.DiffUtil.calculateDiff(DiffUtil.java:145)
        at androidx.recyclerview.widget.DiffUtil.calculateDiff(DiffUtil.java:105)
        at androidx.recyclerview.widget.AsyncListDiffer$1.run(AsyncListDiffer.java:292)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)

**Screenshots**

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/22665789/55274907-0e83bd00-5304-11e9-9362-dd7d3e540d27.gif)

**Smartphone Info:**

|               | |
|---------------|-|
|Device         |Nokia 6.1 Plus|
|Android Version| Pie 9.0    |

**Additional context**

The app seems to crash for certain events and not all. If you are unable to reproduce this issue, try on some other event.

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

**Comment by liveHarshit** [comment]
I think it is memory bounce issue.

**Comment by addiegupta** [comment]
@liveHarshit I am getting this error

2019-03-31 11:54:39.269 8825-9677/com.eventyay.attendee E/AndroidRuntime: FATAL EXCEPTION: pool-1-thread-2
    Process: com.eventyay.attendee, PID: 8825
    java.lang.IndexOutOfBoundsException: Index: 24, Size: 12
        at java.util.ArrayList.get(ArrayList.java:437)
        at androidx.recyclerview.widget.AsyncListDiffer$1$1.areItemsTheSame(AsyncListDiffer.java:306)
        at androidx.recyclerview.widget.DiffUtil.diffPartial(DiffUtil.java:233)
        at androidx.recyclerview.widget.DiffUtil.calculateDiff(DiffUtil.java:145)
        at androidx.recyclerview.widget.DiffUtil.calculateDiff(DiffUtil.java:105)
        at androidx.recyclerview.widget.AsyncListDiffer$1.run(AsyncListDiffer.java:292)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)

GIF:

**Comment by addiegupta** [comment]
I am having a little trouble fixing this. I have narrowed this down to a dependency injection issue specifically the following line of code

`similarEventsListAdapter = get(scope = getOrCreateScope(Scopes.SIMILAR_EVENTS_FRAGMENT.toString()))`

I am not very familiar with dependency injection and after tinkering around with this, what I have assessed is that the same instance of adapter is used for the new SimilarEventsListFragment. And then  on going back to the previous SimilarEventsListFragment, `java.lang.IndexOutOfBoundsException: Index: 20, Size: 1` occurs.

For debugging purposes, replacing the similarEventsListAdapter instantiation as given above with `similarEventsListAdapter= EventsListAdapter(EventLayoutType.SIMILAR_EVENTS, get())` and the crash does not occur as multiple adapters are being used.

How should I proceed?
